### ISSUE TITLE
Why No echo/noise/system sound cancelation?

### ISSUE THREAD

**Issue by abdullah5490** [BearWare/TeamTalk5#1230]
### Description

the "system sound" filteration or "noise filteration" is an essential feature for all communication platforms which protects/filter sounds playing on our system from going in to the ongoing call/session. It is present in almost every communication clients, But teamTalk clients lack this important feature which is a greatest minus point of it. Due to what, all of sounds playing on system is always going directly into the audio session on teamTalk, so that we cannot easily use our computer while remaining on teamTalk session because the screen reader voice would not be filtered from call participants.

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [X] TeamTalkClassic
- [ ] TeamTalkServer

### Platform

- [X] Windows
- [ ] macOS
- [ ] Android
- [ ] iOS
- [X] Linux

### Expected behavior

While we are using headset with 3.5MM combo connector, our system sounds should not be broadcasted along with our voice during teamTalk ongoing session. This could be prevented by implementing "echo cancelation, noise filteration or system sound filteration" in TeamTalk clients. These (or similar) features are essential for every communication platforms and are already working under other options like skype or MS team. IT is really necessary to have such filteration to protect our privacy and avoid disturbance for other participants on a TeamTalk call. 

### Actual behavior

While we are using headset with 3.5MM combo connector, TeamTalk always broadcast all system sounds to the ongoing call along with our voice even the "system sound sharing" is not enabled, and there is no choice to restrict it. Its a privacy breech as well as great disturbance for other participants present in ongoing audio  session. The only way to stop this from happening to keep ourselves muted, which is not a suitable way to be muted everytime while on a call.

### Steps to reproduce problem

1. Pick up a laptop which has a combo audio jack for headset and connect a compatible headset (not headfone) to it. It should recognize the external mik present in your headset.
2. Join a teamTalk audio call/session and use your screen reader, other participants will be hearing the sound of your screen reader.
3. Do the same on other communication software such as skype, the other party on the call will not hear anything except your own voice.

**Comment by AbdullahZubair** [comment]
Indeed, this is one of the major issue, that prevent us to using Team Talk as our primary communication source. even having a dedicated server of TT, but couldn't use that because of such type of issue.
Kindly fix it ASAP.

**Comment by bear101** [comment]
If you try installing [TeamTalk v5.6.3](https://bearware.dk/teamtalk/v5.6.3/) then you are able to echo cancel system sounds. But the sound is reduced to 22050 Hz mono.

**Comment by abdullah5490** [comment]
@bear101  What do you mean by "reduced to 22050 Hz". The feature is capable to filter/cancel the system sound completely if it is coming from same root, not only slite reduction. 
If it was available in V5.6.3, then it will be more than better you could improve it to the next/comprehensive level.

**Comment by abdullah5490** [comment]
still waiting for kind look at this issue

**Comment by ChrisDuffley** [comment]
To @bear101: This is correct, and I hate using it because of that reduction in quality. I've seen Zoom use better Echo Cancellation algorithms though and I do wonder what they're using. If it's any good and can be used somewhere else, perhaps that can be the way to go with it in TT.

Christopher Duffley

Host of Mission Possible Podcast <http://www.missionpossiblepodcast.com/>

LINKS TO SUBSCRIBE:

iTunes <https://itunes.apple.com/us/podcast/mission-possible-christopher/id1103092109?mt=2> | Stitcher <http://www.stitcher.com/s?fid=88094&refid=stpr> | Google Play <https://goo.gl/app/playmusic?ibi=com.google.PlayMusic&isi=691797987&ius=googleplaymusic&link=https://play.google.com/music/m/Ik76dexklcprwnpuxbdr2s6bm3y?t=Mission_Possible_with_Christopher_Duffley> | Follow Podcast On TuneIn And Listen To Latest Episode <http://tunein.com/radio/Mission-Possible-With-Christopher-Duffley-p892505/>

LIKE/FOLLOW Me:

Facebook <http://facebook.com/christopherduffley> | Twitter <http://twitter.com/chrisduffley> | YouTube <http://youtube.com/christopherduffley> | Instagram <http://instagram.com/christopherduffley>

On Feb 13, 2022, at 01:41, Bjørn Damstedt Rasmussen ***@***.***> wrote:


If you try installing TeamTalk v5.6.3 <https://bearware.dk/teamtalk/v5.6.3/> then you are able to echo cancel system sounds. But the sound is reduced to 22050 Hz mono.
